api/envoy/config/core/v3/health_check.proto

```diff
@@ -414,4 +414,9 @@ message HealthCheck {
   // the cluster's :ref:`transport socket <envoy_v3_api_field_config.cluster.v3.Cluster.transport_socket>`
   // will be used for health check socket configuration.
   google.protobuf.Struct transport_socket_match_criteria = 23;
+
+  // When the health status of a host is healthy, if this field is enabled, when envoy wants to send active health check packet to target host, it will first check whether
+  // there is already business traffic hit the target host during the interval. If yes, no need to send health check packet.
+  // The default value is false.
```

Comment:
I will help you copy edit this later, but in the meantime this needs more clarity around how HTTP is handled, gRPC, etc. Also, I think this should also work for other health checkers like Redis?

Additionally, this should have some documentation added in the health checking arch overview page.